site/content/extend/plugins/webapp/reference.md

```diff
@@ -96,3 +111,80 @@ The theme object has the following properties:
 | mentionHighlightLink | Color of text for mention links in posts |
 | codeTheme | Code block theme, either 'github', 'monokai', 'solarized-dark' or 'solarized-light' |
 
+## Exported Libraries and Functions
+
+The web app exports a number of libraries and functions on the [window](https://developer.mozilla.org/en-US/docs/Web/API/Window) object for plugins to use. We recommend importing as many libraries from the window as possible. Below is a list of the exposed libraries and functions:
+
+
+| Library | Description |
+| -------- | ----------- |
+| react | Standard [ReactJS](https://reactjs.org/) library |
+| react-dom | [ReactDOM](https://reactjs.org/docs/react-dom.html) |
+| redux | [Redux](https://redux.js.org/) |
+| react-redux | [React bindings for Redux](https://github.com/reactjs/react-redux) |
+| react-bootstrap | [Bootstrap for React](https://react-bootstrap.github.io/) |
+| post-utils | Post utility functions for common post related tasks |
+
+#### post-utils
+
+Contains the following post utility functions:
+
+##### `formatText(text, options)`
+Performs formatting of text including Markdown, highlighting mentions and search terms and converting URLs, hashtags, @mentions and ~channels to links by taking a string and returning a string of formatted HTML.
+
+* `text` - String of text to format, e.g. a post's message
+* `options` - (Optional) An object containing the following formatting options
+ * `searchTerm` - If specified, this word is highlighted in the resulting HTML. Defaults to nothing.
+ * `mentionHighlight` - Specifies whether or not to highlight mentions of the current user. Defaults to true.
+ * `mentionKeys` - A list of mention keys for the current user to highlight.
+ * `singleline` - Specifies whether or not to remove newlines. Defaults to false.
+ * `emoticons` - Enables emoticon parsing with a data-emoticon attribute. Defaults to true.
+ * `markdown` - Enables markdown parsing. Defaults to true.
+ * `siteURL` - The origin of this Mattermost instance. If provided, links to channels and posts will be replaced with internal links that can be handled by a special click handler.
+ * `atMentions` - Whether or not to render "@" mentions into spans with a data-mention attribute. Defaults to false.
+ * `channelNamesMap` - An object mapping channel display names to channels. If provided, ~channel mentions will be replaced with links to the relevant channel.
+ * `team` - The current team.
+ * `proxyImages` - If specified, images are proxied. Defaults to false.
+
+##### `messageHtmlToComponent(html, isRHS, options)`
+Converts HTML to React components.
+
+* `html` - String of HTML to convert to React components
+* `isRHS` - Boolean indicating if the resulting componets are to be displayed in the right-hand sidebar
```

Comment:
typo: `components`

Also, it's not super clear to me why I'd use this flag. Does passing `isRhs: true` make the components automatically appear in the RHS? Does HTML in the RHS require special treatment?